File: factorial/C++/factorial.cpp
// Compile with g++ factorial.cpp and run with ./a.out
// Recursive function calculating the factorial of 5

#include <iostream>

using namespace std;

int factorial(int n) {
    if (n == 1)
        return 1;
    return n * factorial(n - 1);
}

int main(void) {
    cout << "The factorial of 5 is: " << factorial(5) << endl;
}

File: factorial/Rust/factorial.rs
// Compile using cargo run factorial.rs

fn factorial(n: i32) -> i32 {
    if n == 1 { return 1 }
    else { return n * factorial(n - 1) }
}

fn main() {
    println!("The factorial of 5 is: {}", factorial(5));
}
